lib/io/sqlite/insert.js

```diff
@@ -0,0 +1,82 @@
+import {
+  Attributes,
+  noteFromNoteModel,
+  attrFromAttrModel,
+  Tags,
+  tagFromTagModel
+} from "./models.js";
+import { loadDB } from "./setup.js";
+import { saveNoteContent, executeIfDefined } from "../util.js";
+const Promise = require("bluebird");
+
+function getAttributeModels(attributes, noteID) {
+  const attrs = [];
+  attributes.forEach(attr => {
+    attrs.push(attrFromAttrModel(attr, noteID));
+  });
+  return attrs;
```

Comment:
Not a problem, but you could always use a `map` for this type of pattern. It's very "javascript-y". Example:
``` js
return attributes.map(attr => attrFromAttrModel(attr, noteID));
```

Reply:
Thanks, that's a much cleaner way of doing it!